src/BufferedSocket.php

```diff
@@ -0,0 +1,58 @@
+<?php
+namespace KS;
+
+class BufferedSocket extends BaseSocket
+{
+    private $buffer="";
+    private $socketClosed=false;
+
+    public function processReadReady()
+    {
+        $data = parent::readData();
+        if ($data === false) {
+            $socketClosed = true;
+            return Result::FAILED;
+        }
+        if (\strlen($data)===0) {
+            $this->socketClosed = true;
+        }
+        $this->buffer .= $data;
+        return Result::SUCCEEDED;
+    }
+
+    public function isSocketClosed()
+    {
+        return $this->socketClosed;
+    }
+
+    public function getReadBuffer()
+    {
+        return $this->buffer;
+    }
+
+    public function consumeReadBuffer($bytesToConsume)

```

Comment:
To me, `consume` implies a return. If it doesn't return, it seems more correct to call it `truncate`. Given this, maybe it make sense to eliminate `getAndConsumeReadBuffer` and just have `consumerReadBuffer` return the consumed portion of the buffer?